tests/datasource/fluent/test_sql_datasources.py

```diff
@@ -28,5 +32,114 @@ def test_kwargs_are_passed_to_create_engine(mocker: MockerFixture):
+    """
+    table_names_in_dbms_schema: list[str | sqlalchemy.quoted_name] = [
+        "table_name_0",
+        "table_name_1",
+        "table_name_2",
+        "table_name_3",
+    ]
+
+    with pytest.raises(ValueError) as eee:
+        _ = _verify_table_name_exists_and_get_normalized_typed_name_map(
+            name="nonexistent_table_name",
+            typed_names=table_names_in_dbms_schema,
+        )
+        assert (
+            str(eee.value)
+            == 'Error: The table "nonexistent_table_name" does not exist.'
+        )
+
+
+@pytest.mark.unit
+def test_table_quoted_name_type_all_upper_case_normalizion_is_noop():
+    """
+    DBMS entity names (table, column, etc.) must adhere to correct case insensitivity standards.  All upper case is
+    standard for Oracle, DB2, and Snowflake, while all lowercase is standard for SQLAlchemy; hence, proper conversion to
+    quoted names must occur.  This test ensures that all upper case entity usage does not undergo any conversion.
+    """
+    table_names_in_dbms_schema: list[str | sqlalchemy.quoted_name] = [
+        "ACTORS",
+        "ARTISTS",
+        "ATHLETES",
+        "BUSINESS_PEOPLE",
+        "HEALTHCARE_WORKERS",
+        "ENGINEERS",
+        "LAWYERS",
+        "MUSICIANS",
+        "SCIENTISTS",
+        "LITERARY_PROFESSIONALS",
+    ]
+
+    name: str
+    normalized_table_name_mappings: Sequence[
+        Tuple[str, str | sqlalchemy.quoted_name] | None
+    ] = [
+        _verify_table_name_exists_and_get_normalized_typed_name_map(
+            name=name,
+            typed_names=table_names_in_dbms_schema,
+        )
+        for name in table_names_in_dbms_schema
+    ]
+
+    idx: int
+    normalized_table_name_mapping: Tuple[str, str | sqlalchemy.quoted_name] | None
+    for idx, normalized_table_name_mapping in enumerate(normalized_table_name_mappings):
+        assert normalized_table_name_mapping[0] == normalized_table_name_mapping[1]
+
+
+@pytest.mark.unit
+def test_table_quoted_name_type_all_lower_case_normalizion_full():
+    """
+    DBMS entity names (table, column, etc.) must adhere to correct case insensitivity standards.  All upper case is
+    standard for Oracle, DB2, and Snowflake, while all lowercase is standard for SQLAlchemy; hence, proper conversion to
+    quoted names must occur.  This test ensures that all lower case entity usage undergo conversion to quoted literals.
+    """
+    table_names_in_dbms_schema: list[str | sqlalchemy.quoted_name] = [
+        "actors",
+        "artists",
+        "athletes",
+        "business_people",
+        "healthcare_workers",
+        "engineers",
+        "lawyers",
+        "musicians",
+        "scientists",
+        "literary_professionals",
+    ]
+
+    name: str
+
+    quoted_table_names: list[sqlalchemy.quoted_name] = [
+        sqlalchemy.quoted_name(value=name, quote=True)
+        for name in table_names_in_dbms_schema
+    ]
```

Comment:
Instead of relying on `sqlalchemy` to do this, can we just hardcode the list? We're adding an external dependency and now coupling that with our test here

Reply:
@cdkini I made a change to hardcore the `quoted_table_names` -- however, the only way to do this is through the `sqlalchemy.quoted_name()` method.  Thanks.